src/main/java/com/codeborne/selenide/webdriver/ChromeDriverFactory.java

```diff
@@ -2,88 +2,115 @@
 
 import com.codeborne.selenide.Browser;
 import com.codeborne.selenide.Config;
-import org.apache.commons.lang3.math.NumberUtils;
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.MutableCapabilities;
 import org.openqa.selenium.Proxy;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeDriverService;
 import org.openqa.selenium.chrome.ChromeOptions;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.File;
-import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
+import java.util.regex.Pattern;
 
-class ChromeDriverFactory extends AbstractDriverFactory {
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.regex.Matcher.quoteReplacement;
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+public class ChromeDriverFactory extends AbstractDriverFactory {
   private static final Logger log = LoggerFactory.getLogger(ChromeDriverFactory.class);
 
-  @Override
-  WebDriver create(Config config, Proxy proxy) {
-    ChromeOptions options = createChromeOptions(config, proxy);
-    return new ChromeDriver(options);
-  }
+  // Regexp from https://stackoverflow.com/a/15739087/1110503 to handle commas in values
+  private static final Pattern REGEX_COMMAS_IN_VALUES = Pattern.compile(",(?=([^\"]*\"[^\"]*\")*[^\"]*$)");
+  private static final Pattern REGEX_REMOVE_QUOTES = Pattern.compile("\"", Pattern.LITERAL);
 
   @Override
   boolean supports(Config config, Browser browser) {
     return browser.isChrome();
   }
 
-  ChromeOptions createChromeOptions(Config config, Proxy proxy) {
+  @Override
+  public void setupBinary() {
+    if (isSystemPropertyNotSet("webdriver.chrome.driver")) {
+      WebDriverManager.chromedriver().setup();
+    }
+  }
+
+  @Override
+  @SuppressWarnings("deprecation")
```

Comment:
Will it be good for Selenium 4 migration?

Reply:
When Selenium 4 gets finally released, we will find a solution. :) 

Actually I am going to suggest a fix to `ChromeOptions` class in Selenium 4.